crates/bevy_render/src/primitives/mod.rs

```diff
@@ -49,6 +51,34 @@ impl Aabb {
         }
     }
 
+    /// Returns a bounding box enclosing the specified set of points.
+    ///
+    /// Returns `None` if the iterator is empty.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use bevy_math::{Vec3, Vec3A};
+    /// # use bevy_render::primitives::Aabb;
+    /// let bb = Aabb::enclosing([Vec3::X, Vec3::Z * 2.0, Vec3::Y * -0.5]).unwrap();
+    /// assert_eq!(bb.min(), Vec3A::new(0.0, -0.5, 0.0));
+    /// assert_eq!(bb.max(), Vec3A::new(1.0, 0.0, 2.0));
+    /// ```
+    pub fn enclosing<T: Borrow<Vec3>>(iter: impl IntoIterator<Item = T>) -> Option<Self> {
+        let mut iter = iter.into_iter().map(|p| *p.borrow());
+        let mut min = iter.next()?;
+        let mut max = min;
+        for Vec3 { x, y, z } in iter {
+            min.x = f32::min(min.x, x);
+            min.y = f32::min(min.y, y);
+            min.z = f32::min(min.z, z);
+            max.x = f32::max(max.x, x);
+            max.y = f32::max(max.y, y);
+            max.z = f32::max(max.z, z);
+        }
```

Comment:
```suggestion
       for v in iter {
            min = Vec3::min(min, v);
            max = Vec3::max(max, v);
        }
```

You can use [`Vec3::min`](https://docs.rs/glam/latest/glam/f32/struct.Vec3.html#method.min) and [`Vec3::max`](https://docs.rs/glam/latest/glam/f32/struct.Vec3.html#method.max) these are element-wise min and max operation.